src/components/lab/thermometer/Scale.js

```diff
@@ -0,0 +1,217 @@
+    (key, value) => value * tickStep + roundFrom,
+  );
+
+  // select marks at most number of scale we can display
+  const maxNbScale = Math.floor(thermometerHeight / SCALE_HEIGHT);
+  const prop = Math.ceil(scales.length / maxNbScale);
+  if (prop > 1) {
+    scales = scales.filter((_, i) => i % prop === 0);
+  }
+
+  const heightBetweenTicks = thermometerHeight / scales.length;
+
+  // compute fill height given current temperature value
+  const fillValue = temperatureToHeight({
+    heightBetweenTicks,
+    currentTemperature,
+    tickStep,
+    minTemperature: roundFrom,
+    maxTemperature: roundTo,
+  });
+
+  // draw scale ticks
+  const scaleComponents = scales.map((text, idx) => {
+    const thermometerYPosition =
+      offsetY + thermometerHeight - idx * heightBetweenTicks;
+
+    return (
+      <Group key={text}>
+        <Text
+          x={thermometerXPosition + SCALE_PADDING_LEFT}
+          y={thermometerYPosition - SCALE_FONT_SIZE / 3}
+          text={text}
+          fontSize={SCALE_FONT_SIZE}
+        />
+        <Rect
+          x={THERMOMETER_POSITION_X + THERMOMETER_WIDTH - SCALE_WIDTH}
+          y={thermometerYPosition}
+          width={SCALE_WIDTH}
+          height={SCALE_LINE_HEIGHT}
+          fill={THERMOMETER_STROKE_COLOR}
+        />
+      </Group>
+    );
+  });
+
+  const sliderPositionX =
+    THERMOMETER_POSITION_X +
+    THERMOMETER_WIDTH +
+    SCALE_WIDTH +
+    SCALE_PADDING_LEFT +
+    SCALE_FONT_SIZE * 2;
+
+  const minThermometerHeight = offsetY + thermometerHeight;
+  const maxThermomerterHeight =
+    minThermometerHeight - (scales.length - 1) * heightBetweenTicks;
+  const currentTemperatureY = offsetY + thermometerHeight - fillValue;
+  return (
+    <>
+      {/* current temperature fill */}
+      <Rect
+        fill={THERMOMETER_COLOR}
+        x={THERMOMETER_POSITION_X}
+        y={currentTemperatureY}
+        width={THERMOMETER_WIDTH}
+        height={fillValue}
+      />
+
+      {/* scales */}
+      {scaleComponents}
+
+      {/* triangle slider */}
+      <RegularPolygon
+        draggable
+        dragBoundFunc={(pos) => {
+          // clamp y position
+          let newPositionY = pos.y;
+          if (newPositionY > minThermometerHeight) {
+            newPositionY = minThermometerHeight;
+          } else if (newPositionY < maxThermomerterHeight) {
+            newPositionY = maxThermomerterHeight;
+          }
+          // compute temperature from slider y position
+          const newTemperature = heightToTemperature({
+            heightBetweenTicks,
+            height: newPositionY,
+            tickStep,
+            minTemperature: roundFrom,
+            maxTemperature: roundTo,
+            offsetY,
+            thermometerHeight,
+          });
+
+          dispatchSetTemperature(newTemperature);
+
+          return {
+            x: sliderPositionX,
+            y: newPositionY,
+          };
+        }}
+        x={sliderPositionX}
+        y={currentTemperatureY}
```

Comment:
Should we factor out the following three numbers?

Reply:
Since the slider is a rectangle I don't think factoring out `sides` and `rotation` is necessary. I will however factor out the radius.